splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java

```diff
@@ -1190,6 +1189,7 @@ private static void snapshot(ResultSet rs, String snapshotName, String schemaNam
                     new SnapshotDescriptor(snapshotName, schemaName, objectName, conglomerateNumber,creationTime, null);
             admin.snapshot(sname, "splice:" + conglomerateNumber);
             dd.addSnapshot(descriptor, tc);
+            snapshotList.add(sname);
```

Comment:
not sure how it helps, list add() is only moved down in the same loop, there is no condition